packages/engine-server/src/markdown/remark/backlinksHoverProcessor.ts

```diff
@@ -0,0 +1,232 @@
+import { HTML, Paragraph } from "mdast";
+import Unified, { Transformer } from "unified";
+import { Node, Position } from "unist";
+import visit from "unist-util-visit";
+import { VFile } from "vfile";
+import { DendronASTTypes } from "../types";
+import { RemarkUtils } from "./utils";
+
+/**
+ * Options for the backlinks hover transformer. If using
+ * ProcFlavor.BACKLINKS_PANEL_HOVER, then this must be set.
+ */
+export type BacklinkOpts = {
+  /**
+   * How many lines before and after the backlink to show in the hover
+   */
+  linesOfContext: number;
+
+  /**
+   * The location of the backlink text
+   */
+  location: Position;
+};
+
+/**
+ * Unified processor for rendering text in the backlinks hover control. This
+ * processor returns a transformer that does the following:
+ * 1. Highlights the backlink text
+ * 2. Changes the backlink node away from a wikilink/noteref to prevent the
+ *    backlink text from being altered
+ * 3. Adds contextual " --- line # ---" information
+ * 4. Removes all elements that lie beyond the contextual lines limit of the
+ *    backlink
+ * @param this
+ * @param _opts
+ * @returns
+ */
+export function BacklinkHoverProcessor(
```

Comment:
can we stick with the existing style? so lowerCamelCase, and without using `Processor` suffix?